### ISSUE TITLE
False positive for `attributes` check when we use `@_spi(Something) import Framework`

### ISSUE THREAD

**Issue by hesyifei** [realm/SwiftLint#4868]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

False positive for `attributes` check when we use `@_spi(Something) import Framework`.

Consider the following code:
```swift
@_spi(Something) import Framework

@_spi(SomethingElse)
public struct MyStruct {}
```

According to https://realm.github.io/SwiftLint/attributes.html,
> Attributes should be on their own lines in functions and types, but on the same line as variables and imports

which means that the code above should pass the check.

_Note that this was not an error on 0.49.1._

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ ./swiftlint spi-example.swift
Linting Swift files at paths spi-example.swift
Linting 'spi-example.swift' (1/1)
....../spi-example/spi-example.swift:1:18: warning: Attributes Violation: Attributes should be on their own lines in functions and types, but on the same line as variables and imports (attributes)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
  * 0.51.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
  * `portable_swiftlint` from https://github.com/realm/SwiftLint/releases/download/0.51.0/portable_swiftlint.zip
* Paste your configuration file:
```yml
opt_in_rules:
 - attributes
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.
  * No
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
@_spi(Something) import Framework

/// Hello
@_spi(SomethingElse)
public struct MyStruct {}
```

Command:
```sh
echo "@_spi(Something) import Framework

/// Hello
@_spi(SomethingElse)
public struct MyStruct {}" | ./swiftlint lint --no-cache --use-stdin --enable-all-rules
```
Output:
```
The `anyobject_protocol` rule is now deprecated and will be completely removed in a future release.
The `inert_defer` rule is now deprecated and will be completely removed in a future release due to an equivalent warning issued by the Swift compiler.
The `unused_capture_list` rule is now deprecated and will be completely removed in a future release due to an equivalent warning issued by the Swift compiler.
<nopath>:1:18: warning: Attributes Violation: Attributes should be on their own lines in functions and types, but on the same line as variables and imports (attributes)
Done linting! Found 1 violation, 0 serious in 1 file.
```

**Comment by SimplyDanny** [comment]
This is a duplicate of #4843 and will hopefully be fixed by #4855. Let me know if the new option proposed in the PR would be satisfactory for you.
